src/freertos_drivers/esp32/Esp32WiFiClientAdapter.hxx

```diff
@@ -122,6 +123,34 @@ public:
         return bytesRead;
     }
 
+    /// @return true if the underlying WiFiClient is still connected.
+    bool connected()
+    {
+        return client_.connected();
+    }
+
+    /// @return remote IP address from the underlying WiFiClient.
+    IPAddress remoteIP() const
+    {
+        return client_.remoteIP();
+    }
+
+    /// @return remote port from the underlying WiFiClient.
+    uint16_t remotePort() const
+    {
+        return client_.remotePort();
+    }
+
+    /// @return true if we successfully reconnected to the remote connection
+    bool reconnect()
+    {
+        if (connected())
+        {
+            return true;
+        }
+        return client_.connect(client_.remoteIP(), client_.remotePort());
```

Comment:
you are probably right, these should get cached upon creation of the adapter so they can be reused later instead of relying on the underlying LwIP stack to provide meaningful values at all times.